lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+  
+  if array1.length == array2.length
+    loop_count = array1.length
+    loop_count.times do |i|
+      if array1[i] != array2[i]
+          return false
+      else
```

Comment:
Style nit: do we need this `else` block? I have a feeling we can delete it, if we put a `return true` statement after it.

(`return`ing exits the function - so any code after an executed `return` statement won't run.)